src/main/java/keycontacts/storage/JsonAdaptedStudent.java

```diff
@@ -109,14 +113,22 @@ public Student toModelType() throws IllegalValueException {
 
         final Set<Tag> modelTags = new HashSet<>(studentTags);
 
+        if (gradeLevel == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                GradeLevel.class.getSimpleName()));
+        }
+
+        final GradeLevel modelGradeLevel = new GradeLevel(gradeLevel);
```

Comment:
this section will also need the `if (!GradeLevel.isValidGradeLevel(gradeLevel))...`